### ISSUE TITLE
AMP training on BYOL leads to nan loss in linear eval

### ISSUE THREAD

**Issue by SymbolZ** [open-mmlab/mmselfsup#82]
With AMP, I trained BYOL on 2 machines with 8GPUs, 16GPUs in total. I did not change anything except the batchsize of a GPU from 32 to 128. **Linear eval** encountered nan loss.
Without AMP(same environment and same config file removing AMP), the linear eval produced a normal loss.

I wonder if someone else ran into this problem. Is it the incompatibility between BYOL and AMP, or just that I did something wrong.

**Comment by XiaohangZhan** [comment]
It is weird that BYOL is not compatible with AMP. Does it occur with batchsize=32 * 16?

**Comment by xieenze** [comment]
Hi, Do you use AMP in linear evaluation or in pretraining? 
BYOL pretraining+ AMP works. see [this config](https://github.com/open-mmlab/OpenSelfSup/blob/master/configs/selfsup/byol/r50_bs2048_accumulate2_ep200_fp16.py)
We do not encourage using AMP on Linear evaluation.

**Comment by SymbolZ** [comment]
It seem that I didn't turn on the AMP correctly. This line is missing. 
`optimizer_config = dict(update_interval=update_interval, use_fp16=use_fp16)
`
